### PR TITLE
Refactoring the utilities to reduce monosize of components 

### PR DESCRIPTION
**Pull request by srmukher**
Refactoring the utilities to remove indirect function invocations. This reduced the monosize of each cartesian components by 4kB on an average.
image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-charting-4903549d-4b4c-4971-bfce-c488ad145a47.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-4903549d-4b4c-4971-bfce-c488ad145a47.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31291&commitId=3608156aa0553010661b60fe98e0a38b2030d6b4&env=prod&clientType=FLUENTUI)


  <details><summary><b>react-charting-VerticalBarChart</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-VerticalBarChart.Basic - Secondary Y Axis.default.chromium.png|474| Changed|
|react-charting-VerticalBarChart.Basic - Secondary Y Axis - Dark Mode.default.chromium.png|495| Changed|
|react-charting-VerticalBarChart.Basic - Secondary Y Axis - RTL.default.chromium.png|474| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/components/AreaChart/__snapshots__/AreaChart.test.tsx.snap [hunk 2]
```diff
@@ -1157,7 +1146,6 @@ exports[`AreaChart - mouse events Should render customized callout per stack on
           fill="transparent"
           id="0-line-areaChart_0"
           onMouseMove={[Function]}
-          onMouseOut={[Function]}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/AreaChart/__snapshots__/AreaChart.test.tsx.snap:1160]: why has the snapshot changed

File: packages/react-charting/src/components/CommonComponents/CartesianChart.types.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/CommonComponents/CartesianChart.types.ts:1]: why is this disable line needed. Remove and fix the explicit any.

File: packages/react-charting/src/components/CommonComponents/CartesianChart.base.tsx [hunk 4]
```diff
@@ -424,7 +402,6 @@ export class CartesianChartBase extends React.Component<IModifiedCartesianChartP
           yScale,
           this.props.noOfCharsToTruncate || 4,
           this.props.showYAxisLablesTooltip || false,
-          this.state.startFromX,
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/CommonComponents/CartesianChart.base.tsx:427]: why this change

Comment (srmukher) [reply packages/react-charting/src/components/CommonComponents/CartesianChart.base.tsx:427]: this argument was unused

File: packages/react-charting/src/components/CommonComponents/CartesianChart.base.tsx [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/CommonComponents/CartesianChart.base.tsx:1]: why moving no explicit any to top of file.

Comment (srmukher) [reply packages/react-charting/src/components/CommonComponents/CartesianChart.base.tsx:1]: because there were multiple line wise any. Files with a single line any disabled, did not make any change like MultiStackedBarChart.

Comment (AtishayMsft) [reply packages/react-charting/src/components/CommonComponents/CartesianChart.base.tsx:1]: what is the difference between the 2 representations. Which is better?

**Comment by AtishayMsft** [comment]
Can you update the PR description to represent only the changes in this PR.

File: change/@fluentui-react-charting-4903549d-4b4c-4971-bfce-c488ad145a47.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+  "type": "none",
```
Comment (AtishayMsft) [inline change/@fluentui-react-charting-4903549d-4b4c-4971-bfce-c488ad145a47.json:2]: this should be a patch change

File: packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx [hunk 7]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx:1]: revert change

File: packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx [hunk 8]
```diff
@@ -207,6 +216,32 @@ export class GroupedVerticalBarChartBase extends React.Component<
...
+    chartType: ChartTypes,
+    isRTL: boolean,
+    xAxisType: XAxisTypes,
+    barWidth: number,
+    tickValues: Date[] | number[] | undefined,
+    shiftX: number,
+  ) => {
+    let domainNRangeValue: IDomainNRange;
+    if (xAxisType === XAxisTypes.NumericAxis) {
+      domainNRangeValue = { dStartValue: 0, dEndValue: 0, rStartValue: 0, rEndValue: 0 };
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx:236]: combine these 2 if/else statements

File: packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx [hunk 9]
```diff
@@ -173,8 +178,12 @@ export class GroupedVerticalBarChartBase extends React.Component<
         calloutProps={calloutProps}
         legendBars={legends}
         xAxisType={this._xAxisType}
+        createYAxis={createYAxisForOtherCharts}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx:181]: rename this function

File: packages/react-charting/src/utilities/UtilityUnitTests.test.ts [hunk 10]
```diff
@@ -970,180 +933,6 @@ describe('domainRageOfVerticalNumeric', () => {
   });
 });
 
-describe('getDomainNRangeValues', () => {
```
Comment (AtishayMsft) [inline packages/react-charting/src/utilities/UtilityUnitTests.test.ts:973]: why remove test cases?

File: packages/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx [hunk 11]
```diff
@@ -215,6 +231,32 @@ export class HeatMapChartBase extends React.Component<IHeatMapChartProps, IHeatM
...
+    chartType: ChartTypes,
+    isRTL: boolean,
+    xAxisType: XAxisTypes,
+    barWidth: number,
+    tickValues: Date[] | number[] | undefined,
+    shiftX: number,
+  ) => {
+    let domainNRangeValue: IDomainNRange;
+    if (xAxisType === XAxisTypes.NumericAxis) {
+      domainNRangeValue = { dStartValue: 0, dEndValue: 0, rStartValue: 0, rEndValue: 0 };
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx:251]: same here

File: packages/react-charting/src/utilities/UtilityUnitTests.test.ts (hunk 10 above)
Comment (srmukher) [reply packages/react-charting/src/utilities/UtilityUnitTests.test.ts:973]: Removed the test cases for the functions which were removed as the original functions were invoked. The tests for the original functions they call are still there.

File: packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx (hunk 9 above)
Comment (srmukher) [reply packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx:181]: renaming it to createYAxisForCharts

File: packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx (hunk 8 above)
Comment (srmukher) [reply packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx]: done

File: packages/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx (hunk 11 above)
Comment (srmukher) [reply packages/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx]: done

File: packages/react-charting/src/utilities/utilities.ts [hunk 12]
```diff
@@ -359,7 +359,7 @@ export function createYAxisForHorizontalBarChartWithAxis(yAxisParams: IYAxisPara
   return yAxisScale;
 }
 
-export function createYAxisForOtherCharts(
+export function createYAxisForCharts(
```
Comment (AtishayMsft) [inline packages/react-charting/src/utilities/utilities.ts:362]: call this function createNumericYAxis

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 13]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:1]: revert blanket eslint disable statement

File: packages/react-charting/src/utilities/UtilityUnitTests.test.ts (hunk 10 above)
Comment (AtishayMsft) [reply packages/react-charting/src/utilities/UtilityUnitTests.test.ts:973]: Can you share the corresponding tests for the scenario that you have removed

Comment (srmukher) [reply packages/react-charting/src/utilities/UtilityUnitTests.test.ts:973]: The functions that say 'getDomainNRangeValues' function called earlier were: domainRangeOfNumericForAreaChart
domainRangeOfVSBCNumeric
domainRageOfVerticalNumeric
domainRangeOfNumericForHorizontalBarChartWithAxis
domainRangeOfDateForAreaLineVerticalBarChart
domainRangeOfXStringAxis

All tests of these internal functions exist in the same file.

File: packages/react-charting/src/utilities/utilities.ts (hunk 12 above)
Comment (srmukher) [reply packages/react-charting/src/utilities/utilities.ts]: done

**Comment by Thanneerudeepika** [comment]
Performed Full test pass on this PR https://fluentuipr.z22.web.core.windows.net/pull/31291/public-docsite/index.html#/controls.
Tested the functionality (manual testing) for all the charts.

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 14]
```diff
@@ -187,12 +193,16 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
...
           calloutProps={calloutProps}
           legendBars={legends}
+          createYAxis={createNumericYAxis}
           isCalloutForStack
           xAxisType={isXAxisDateType ? XAxisTypes.DateAxis : XAxisTypes.NumericAxis}
           tickParams={tickParams}
           maxOfYVal={stackedInfo.maxOfYVal}
           getGraphData={this._getGraphData}
+          getDomainNRangeValues={this._getDomainNRangeValues}
+          createStringYAxis={createStringYAxisForOtherCharts}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:203]: this function is still not renamed

File: packages/react-charting/src/utilities/UtilityUnitTests.test.ts (hunk 10 above)
Comment (srmukher) [reply packages/react-charting/src/utilities/UtilityUnitTests.test.ts:973]: For createYAxis: (removed), following are the internal functions whose tests are still there:
createYAxisForHorizontalBarChartWithAxis
createYAxisForOtherCharts-> renamed to createNumericYAxis

For createStringYAxis: (removed), following are the internal functions whose tests are still there:
createStringYAxisForHorizontalBarChartWithAxis
createStringYAxisForOtherCharts

For getDomainNRangeValues (removed), following are the internal functions whose tests are still there:

domainRangeOfNumericForAreaChart
domainRangeOfVSBCNumeric
domainRageOfVerticalNumeric
domainRangeOfNumericForHorizontalBarChartWithAxis
domainRangeOfDateForAreaLineVerticalBarChart
domainRangeOfXStringAxis


For getMinMaxOfYAxis(removed), following are the internal functions whose tests are still there:

findNumericMinMaxOfY
findVSBCNumericMinMaxOfY
findVerticalNumericMinMaxOfY
findHBCWANumericMinMaxOfY

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx (hunk 14 above)
Comment (srmukher) [reply packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:203]: renamed createStringYAxisForOtherCharts to createStringYAxis

File: packages/react-charting/src/utilities/UtilityUnitTests.test.ts (hunk 10 above)
Comment (AtishayMsft) [reply packages/react-charting/src/utilities/UtilityUnitTests.test.ts:973]: for minmax of y HeatMap case is missing
for domainNrange GroupedVerticalNumeric case is missing

Comment (srmukher) [reply packages/react-charting/src/utilities/UtilityUnitTests.test.ts:973]: Updated tests
